src/taskgraph/util/taskcluster.py

```diff
@@ -494,16 +453,14 @@ def get_ancestors(
         task_ids = [task_ids]
 
     for task_id in task_ids:
-        try:
-            task_def = get_task_definition(task_id, use_proxy)
-        except requests.HTTPError as e:
+        task_def = get_task_definition(task_id)
+        if not task_def:
             # Task has most likely expired, which means it's no longer a
             # dependency for the purposes of this function.
-            if e.response.status_code == 404:
-                continue
-
-            raise e
+            continue
```

Comment:
Are you sure the taskcluster client would return `None` on a 404? I would have thought it would raise an exception still.. Though it would probably be from aiohttp rather than requests.